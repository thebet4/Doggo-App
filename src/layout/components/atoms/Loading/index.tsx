import React from "react";
import { Spinner } from "./styles";

interface ILoadingProps {
  size?: "small" | "large";
}

const Loading = (
  { size }: ILoadingProps = {
    size: "large",
  }
) => {
  return <Spinner size={size} />;
};

export default Loading;
