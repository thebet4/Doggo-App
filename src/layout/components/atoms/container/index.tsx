import React, { ReactElement } from "react";
import { SafeArea } from "./styles";
import { StatusBar } from "expo-status-bar";

type ContainerProps = {
  children: ReactElement;
};

const Container = ({ children }: ContainerProps): JSX.Element => {
  return (
    <SafeArea>
      <StatusBar style="dark" />
      {children}
    </SafeArea>
  );
};
export default Container;
