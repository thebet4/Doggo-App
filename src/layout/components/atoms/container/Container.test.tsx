import React from "react";
import renderer from "react-test-renderer";

import { Container } from "@layout/components/atoms";
import { Text } from "react-native";

it(`renders <Container /> correctly`, () => {
  const tree = renderer
    .create(
      <Container>
        <Text>Lorem ipsom</Text>
      </Container>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
