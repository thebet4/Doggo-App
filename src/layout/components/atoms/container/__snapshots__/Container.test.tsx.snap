// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Container /> correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 16,
        "paddingLeft": 40,
        "paddingRight": 40,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text>
    Lorem ipsom
  </Text>
</RNCSafeAreaView>
`;
